src/anomalib/utils/metrics/aupro.py

```diff
@@ -96,9 +107,29 @@ def compute_pro(self, cca: Tensor, target: Tensor, preds: Tensor) -> tuple[Tenso
         Returns:
             tuple[Tensor, Tensor]: tuple containing final fpr and tpr values.
         """
+        if self.num_thresholds is not None:
+            # binary_roc is applying a sigmoid on the predictions before computing the roc curve
+            # when some predictions are out of [0, 1], the binning between min and max predictions
+            # cannot be applied in that case. This can be removed when
+            #  https://github.com/Lightning-AI/torchmetrics/issues/1526 is fixed and
+            #  the roc curve is computed with deactivated formatting.
+
+            if torch.all((0 <= preds) * (preds <= 1)):
+                thresholds = thresholds_between_min_and_max(preds, self.num_thresholds, self.device)
+            else:
+                thresholds = thresholds_between_0_and_1(self.num_thresholds, self.device)
+
+        else:
+            thresholds = None
 
         # compute the global fpr-size
-        fpr: Tensor = roc(preds, target)[0]  # only need fpr
+        fpr: Tensor = binary_roc(
+            preds=preds,
+            target=target,
+            thresholds=thresholds,
+        )[
+            0
+        ]  # only need fpr
```

Comment:
Can you use the `roc` function here instead? The reason is that `binary_roc` maps the predictions to the [0, 1] range using sigmoid, which is exactly what we're trying to avoid. We asked the TorchMetrics developers to make the sigmoid mapping optional, but until then it would be better if we use the legacy `roc` function, which does not remap the predictions.

Reply:
indeed, the roc call with thresholds :
- is requiring the task to be defined. Otherwise the thresholds will just be silently ignored.
- is automatically calling binary roc at the end.

This roc function will be soon deprecated so I would suggest to keep it like this.

Also @djdameln we added a comment just above to mention this issue.